### PR TITLE
[Serving] Add path and query parameter support to API handler

### PR DESCRIPTION
**Pull request by jond01**
## Summary

This PR implements path and query parameter support for the serving graph API handler, enabling more flexible and RESTful API endpoint definitions.

## Changes

- **_RequestContext**: Unified parameter container with automatic conflict detection across path, query, and body parameters
- **_parse_query_params**: Static method for parsing URL query strings with support for repeated parameters (returned as lists)
- **_extract_query_params**: Helper method to extract query parameters from both mock events and Nuclio events
- **_compile_endpoint_patterns**: Pre-compiles regex patterns from path templates for performance optimization
- **Type hints**: Added proper type hints for event parameters (`Union[nuclio_sdk.Event, MockEvent]`)
- **Comprehensive test coverage**: 98 tests total (+21 new tests covering all new helper functions)
- **System test**: Integration test for path and query parameter extraction

### Path Parameters
```python
config.add_endpoint_handler("/api/users/{user_id}", HTTPMethod.GET, APIHandlerAction.ALLOW)
# Request: GET /api/users/123
# Handler receives: {"user_id": "123"}
```

### Query Parameters
```python
config.add_endpoint_handler("/api/search", HTTPMethod.GET, APIHandlerAction.ALLOW)
# Request: GET /api/search?q=test&limit=10
# Handler receives: {"q": "test", "limit": "10"}
```

### Repeated Query Parameters
```python
# Request: GET /api/items?id=1&id=2&id=3
# Handler receives: {"id": ["1", "2", "3"]}
```

### Combined Path and Query Parameters
```python
config.add_endpoint_handler("/api/users/{user_id}/items", HTTPMethod.GET, APIHandlerAction.ALLOW)
# Request: GET /api/users/123/items?filter=active&limit=10
# Handler receives: {"user_id": "123", "filter": "active", "limit": "10"}
```

## Testing

- ✅ All 98 unit tests passing
- ✅ System test included (path + query params in real Nuclio deployment)
- ✅ Conflict detection tested (ensures no parameter name collisions)
- ✅ URL encoding handled correctly
- ✅ Edge cases covered (empty values, special characters, etc.)

## Known Limitations

**Nuclio Query Parameter Handling**: The Nuclio Python SDK has a limitation where repeated query parameters only preserve the last value instead of all values. This is tracked in **NUC-745** and assigned to the Nuclio team.

## Related Issues

- Implements: [ML-11657](https://iguazio.atlassian.net/browse/ML-11657)
- Related: [NUC-745](https://iguazio.atlassian.net/browse/NUC-745) (Nuclio SDK limitation)

## Documentation

The feature is documented in code with comprehensive docstrings and examples in the test suite.

[ML-11657]: https://iguazio.atlassian.net/browse/ML-11657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/serving/test_api_handler.py [hunk 1]
```diff
@@ -53,6 +54,329 @@ def do(self, event: MockEvent | str | dict):
...
+        """Test path template with multiple parameters"""
+        config = APIHandlerConfig()
+        config.add_endpoint_handler(
+            "/orgs/{org_id}/repos/{repo_id}/issues/{issue_id}",
+            HTTPMethod.GET,
+            APIHandlerAction.ALLOW,
+        )
+
+        step = _APIHandlerStep(config=config)
+        _, pattern, _, _ = step._endpoint_patterns[0]
```
Comment (assaf758) [inline tests/serving/test_api_handler.py]: some of the test do not have `assert len(step._endpoint_patterns)` - suggest to add.
also, is there a case with len > 1?

File: mlrun/serving/api_handler.py [hunk 2]
```diff
@@ -135,42 +318,58 @@ def do(self, event):
...
                 endpoints_count=len(self.config._endpoints),
             )
 
             # Find matching endpoint
-            matching_endpoint_key = self._match_endpoint(method, path)
+            matching_endpoint_key, path_params = self._match_endpoint(
+                method, normalized_path
+            )
+
+            mlrun.utils.logger.debug(
```
Comment (assaf758) [inline mlrun/serving/api_handler.py:324]: suggest you unify this debug log with the one before calling _match_endpoint

**Review by assaf758 — COMMENTED** [review]
Very nice Jonathan!
Some minor comments inline.

File: mlrun/serving/api_handler.py (hunk 2 above)
Comment (jond01) [reply mlrun/serving/api_handler.py:324]: Merged them

File: tests/serving/test_api_handler.py (hunk 1 above)
Comment (jond01) [reply tests/serving/test_api_handler.py]: Added this assertion.
The cases with 2 endpoints are:
1. `test_compile_endpoint_patterns_with_templates` - same path, different methods (GET and POST).
2. `test_multiple_path_parameters` - added another endpoint to the test.

**Review by assaf758 — APPROVED** [review]
